forecasting/src/autogluon/forecasting/predictor.py

```diff
@@ -0,0 +1,420 @@
+        https://ts.gluon.ai/api/gluonts/gluonts.evaluation.metrics.html
+    path: str, default = None
+        Path to directory where models and intermediate outputs should be saved. If unspecified, a timestamped folder
+        "AutogluonModels/ag-[TIMESTAMP]" will be created in the working directory to store all models.
+    verbosity : int, default = 2
+        Verbosity levels range from 0 to 4 and control how much information is printed to stdout. Higher levels
+        correspond to more detailed print statements, and verbosity=0 suppresses output including warnings.
+        If using `logging`, you can alternatively control amount of information printed via `logger.setLevel(L)`,
+        where `L` ranges from 0 to 50 (Note: higher values of `L` correspond to fewer print statements,
+        opposite of verbosity levels).
+
+    Other Parameters
+    ----------------
+    learner_type : AbstractLearner, default = ForecastingLearner
+        A class which inherits from `AbstractLearner`. The learner specifies the inner logic of the ForecastingPredictor
+        for training models and preprocessing data.
+    learner_kwargs : dict, default = None
+        Keyword arguments to send to the learner (for advanced users only). Options include `trainer_type`, a
+        class inheriting from `AbstractTrainer` which controls training of multiple models. If `path` and `eval_metric`
+        are re-specified within `learner_kwargs`, these are ignored.
+
+    Attributes
+    ----------
+    target_column: str
+        Name of column in training/validation data that contains the target time-series value to be predicted. If
+        not specified explicitly during `fit()`, this will default to "target".
+    """
+
+    predictor_file_name = "predictor.pkl"
+
+    def __init__(
+        self,
+        eval_metric: Optional[str] = None,
+        path: Optional[str] = None,
+        verbosity: int = 2,
+        **kwargs,
+    ):
+        self.verbosity = verbosity
+        set_logger_verbosity(self.verbosity, logger=logger)
+        self.path = setup_outputdir(path)
+
+        learner_type = kwargs.pop("learner_type", ForecastingLearner)
+        learner_kwargs = kwargs.pop("learner_kwargs", dict())
+
+        learner_kwargs.pop("eval_metric", None)
+        learner_kwargs.pop("path", None)
+        self.eval_metric = eval_metric
+        self._learner: AbstractLearner = learner_type(
+            path_context=self.path, eval_metric=eval_metric, **learner_kwargs
+        )
+        self._learner_type = type(self._learner)
+
+        self.target_column = "target"
+
+    @property
+    def _trainer(self) -> AbstractForecastingTrainer:
+        return self._learner.load_trainer()  # noqa
+
+    @apply_presets(FORECASTING_PRESETS_CONFIGS)
+    def fit(
+        self,
+        train_data: TimeSeriesDataFrame,
+        prediction_length: int = 1,
+        target_column: str = "target",
```

Comment:
Wondering why this is specified in fit and not in init? Both are probably valid, but just wondering if there is a specific reason why it should be in fit.

Also: Any reason why this isn't called 'label' to align with the other modules?

Reply:
I'll move this and `prediction_length` to init args for both Predictor and Learner. I'll revise the name to `target` to echo Tabular's style however I'd like to refrain from using `label` as in forecasting we use the "past target" to forecast the "future target" so it's not a 1:1 analogy with label.

Reply:
thanks for explaination! `target` is ok, although we may consider adding an alias `label` which maps to `target` and warns the user to use target instead in case users are coming in from other modules